### PR TITLE
Update application class - Closes #6629 and #6892

### LINKED ISSUE #6629: Update application class to follow new framework architecture
### Description

- Update default application to return an object below, so that external library can use api as dependency.
```
interface DefaultApplication {
 app: Application;
 api: {
   token: xxx,
   dpos: xxx,
  ....
 }, 
}
```
- Update `run` function to accept genesis block as instance of Block.
- 

With the new architecture, external module will need direct access to API for dependency with type. Therefore, we need to expose the default module api as well.

> Update `run` function to accept genesis block as instance of Block.

Genesis block cannot be maintained in the memory. Therefore, it should be passed in the run phase, and only temporally used. (no assignment as variable). Also, receiving as instance will increase the flexibility of the genesis block format.

### Additional Information
- this issue should be handle after #6612

### LINKED ISSUE #6892: Fix `getGeneratorsBetweenTimestamps` type and return keys
### Description

- Update the return type to be `Record<string, number>` to reflect the correct typing
- Result key should be in `binary` format than `utf8`. ie) `generatorList.addresses.map(buf => buf.toString('binary'));`  because some character in Buffer cannot be represented in `utf8` format

### Acceptance Criteria

- Update unit test
- DPoS dependency type should be updated

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #6629 and resolves #6892

### How was it solved?

- Update application class to return default application with APIs
- Remove getDefaultModules since it is no longer useful
- Fix #6892 to match type for build
- Expose bftAPI and validatorAPI
- Fix expected interface name `isSeedRevealValid` from `isValidSeedReveal`
- Fix DPoS not to use config in `addDependencies`

### How was it tested?

- Update unit tests

### CODE REVIEW

File: framework/src/modules/token/api.ts [hunk 1]
```diff
@@ -158,4 +158,57 @@ export class TokenAPI extends BaseAPI {
...
+		} catch (error) {
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
+			recipient = {
+				availableBalance: BigInt(0),
+				lockedBalances: [],
+			};
+		}
+		if (recipient.availableBalance < amount + this._minBalance) {
```
Comment (Incede) [inline framework/src/modules/token/api.ts:202]: Why this check here ?
Do you mean `if (amount < this_minbalance)` ?

**Review by Incede — CHANGES_REQUESTED** [review]

**Review by mehmetegemen — APPROVED** [review]
Great work Shu

File: framework/src/modules/token/api.ts (hunk 1 above)
Comment (shuse2) [reply framework/src/modules/token/api.ts:202]: good catch. it shouldve been `recipient.availableBalance + amount  < + this._minBalance` to check if the  recipient  will have more  than min balance after receiving  the  token

File: framework/src/modules/token/api.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export class TokenAPI extends BaseAPI {
 				lockedBalances: [],
 			};
 		}
-		if (recipient.availableBalance < amount + this._minBalance) {
+		if (recipient.availableBalance + amount < this._minBalance) {
 			throw new Error(
 				`Recipient ${recipientAddress.toString(
 					'hex',
```
Comment (Incede) [inline framework/src/modules/token/api.ts:87]: I think the error message doesn't fit the logic checked here

File: framework/src/modules/token/api.ts [hunk 3]
```diff
@@ -158,4 +158,57 @@ export class TokenAPI extends BaseAPI {
...
+			}
+			recipient = {
+				availableBalance: BigInt(0),
+				lockedBalances: [],
+			};
+		}
+		if (recipient.availableBalance + amount < this._minBalance) {
+			throw new Error(
+				`Recipient ${address.toString(
+					'hex',
```
Comment (Incede) [inline framework/src/modules/token/api.ts:205]: same here, error msg needs to be updated

**Review by Incede — CHANGES_REQUESTED** [review]

**Review by Incede — APPROVED** [review]

**Review by mehmetegemen — APPROVED** [review]

**Pull request merged**